src/crepe/system/CollisionSystem.cpp

```diff
@@ -1,5 +1,393 @@
+}
+
+std::optional<std::pair<std::reference_wrapper<Transform>, std::reference_wrapper<Rigidbody>>>
+CollisionSystem::get_active_transform_and_rigidbody(game_object_id_t game_object_id) {
+    RefVector<Transform> transforms = this->component_manager.get_components_by_id<Transform>(game_object_id);
+    if (transforms.empty()) return std::nullopt;
+
+    RefVector<Rigidbody> rigidbodies = this->component_manager.get_components_by_id<Rigidbody>(game_object_id);
+    if (rigidbodies.empty()) return std::nullopt;
+
+    Transform& transform = transforms.front().get();
+    if (!transform.active) return std::nullopt;
+
+    Rigidbody& rigidbody = rigidbodies.front().get();
+    if (!rigidbody.active) return std::nullopt;
+
+    // Return the active components
+    return std::make_pair(std::ref(transform), std::ref(rigidbody));
+}
+
+CollisionSystem::CollisionInternalType CollisionSystem::check_collider_type(const collider_variant& collider1,const collider_variant& collider2){
+	if(std::holds_alternative<std::reference_wrapper<CircleCollider>>(collider1)){
+		if(std::holds_alternative<std::reference_wrapper<CircleCollider>>(collider2))
+		{
+			return CollisionInternalType::CIRCLE_CIRCLE;
+		}
+		else {
+			return CollisionInternalType::CIRCLE_BOX;
+		}
+	}
+	else {
+		if(std::holds_alternative<std::reference_wrapper<CircleCollider>>(collider2))
+		{
+			return CollisionInternalType::BOX_CIRCLE;
+		}
+		else {
+			return CollisionInternalType::BOX_BOX;
+		}
+	}
+}
+
+bool CollisionSystem::check_collision(const CollisionInternal& first_info,const CollisionInternal& second_info, CollisionInternalType type){
+	switch (type) {
+		case CollisionInternalType::BOX_BOX:	{
+			const BoxCollider & box_collider1 = std::get<std::reference_wrapper<BoxCollider>>(first_info.collider);
+			const BoxCollider & box_collider2 = std::get<std::reference_wrapper<BoxCollider>>(second_info.collider);
+			return check_box_box_collision(box_collider1,box_collider2,first_info.transform,second_info.transform,second_info.rigidbody,second_info.rigidbody);
+		}
+		case CollisionInternalType::BOX_CIRCLE: {
+			const BoxCollider & box_collider = std::get<std::reference_wrapper<BoxCollider>>(first_info.collider);
+			const CircleCollider & circle_collider = std::get<std::reference_wrapper<CircleCollider>>(second_info.collider);
+			return check_box_circle_collision(box_collider,circle_collider,first_info.transform,second_info.transform,second_info.rigidbody,second_info.rigidbody);
+		}
+		case CollisionInternalType::CIRCLE_CIRCLE:	{
+			const CircleCollider & circle_collider1 = std::get<std::reference_wrapper<CircleCollider>>(first_info.collider);
+			const CircleCollider & circle_collider2 = std::get<std::reference_wrapper<CircleCollider>>(second_info.collider);
+			return check_circle_circle_collision(circle_collider1,circle_collider2,first_info.transform,second_info.transform,second_info.rigidbody,second_info.rigidbody);
+		}
+		case CollisionInternalType::CIRCLE_BOX:	{
```

Comment:
Misschien een domme vraag maar is er een verschil tussen een `CIRCLE_BOX` en `BOX_CIRCLE` collision? Maakt het uit of de volgorde van de colliders in de event behouden blijft?

Reply:
Geen verschil naast de volgorde van hoe ze in de pair staan. Met de nieuwe code is dit wat duidelijker. (I kan dan de waardes omswitchen voor 1 functie)

Reply:
Ik ga dit veranderen naar this and other wat vaaker wordt gebruikt.